app/models/book.rb

```diff
@@ -3,9 +3,10 @@ class Book < ApplicationRecord
   has_many :reviews
   has_many :authors, through: :book_authors
 
-  validates_presence_of :title
-  validates_presence_of :page_count
-  validates_presence_of :year_published
+  validates_presence_of :title, :page_count, :year_published
+  validates_uniqueness_of :title
+  validates :year_published, numericality: { only_integer: true, greater_than_or_equal_to: -4000, less_than_or_equal_to: 2019 }
```

Comment:
Meh, why restrict to <= 2019? Then users in 2020 couldn't add books that were written then. I'd say delete that restriction or do `Time.now.year`.

Reply:
Great point. Corrected on branch.